source/keymanweb.js

```diff
@@ -1040,19 +1040,43 @@
       var ipList=document.getElementsByTagName(k==0?'INPUT':'TEXTAREA');
       for(var n=0;n<ipList.length;n++) 
       {      
-        if(ipList[n].className.indexOf('kmw-disabled') < 0 && !ipList[n].readOnly )
-          keymanweb.inputList.push(ipList[n]);
-        if(ipList[n].className) 
-          ipList[n].className=ipList[n].className+' keymanweb-font';
-        else
-          ipList[n].className='keymanweb-font';
+        // Registers all relevant static elements on the page.
+        keymanweb.setupDesktopElement(ipList[n]);
       }
     }
-    //TODO: sort list by y, x position on page
-    
   }  
 
   /**
+   * Function     setupDesktopElement
+   * Scope        Private
+   * Description  Setup one element for non-touch devices and add it to the inputList if it is an input element (desktop browsers).
+   *              Only returns true if the element is a valid input for keymanweb and it is not presently tracked as an input element.
+   * @return   {boolean}
+   */       
+  keymanweb.setupDesktopElement = function(Pelem)
+  { 
+      var lcTagName = Pelem.tagName.toLowerCase();
+      // If it's not one of these, we don't need to hook the OSK into it.
+      if(!(lcTagName == "input" || lcTagName == "textarea")) return false;
+
+      // TODO:  Fix potential issue - We might have an issue if, for some reason, an element is re-added later.
+      // (We may need to ensure we don't re-add an element to keymanweb.inputList that isn't already on it!)
+
+      if(Pelem.className.indexOf('kmw-disabled') < 0 && !Pelem.readOnly)
+      {
+        for(i = 0; i < keymanweb.inputList.length; i++)
```

Comment:
Sure thing.  I thought I saw the pattern I used elsewhere, and I'm still getting used to JavaScript.